### ISSUE TITLE
Notebooks MethodError: length

### ISSUE THREAD

**Issue by tigre200** [mitmath/18S191#143]
I have started following the course, and read through the first two notebooks. In both I get the following error that disables some of the code examples in the notebooks:

```
Failed to show value:

MethodError: no method matching length(::Nothing)

Closest candidates are:

length(!Matched::BitSet)

@ Base bitset.jl:348

length(!Matched::Base.AsyncGenerator)

@ Base asyncmap.jl:390

length(!Matched::RegexMatch)

@ Base regex.jl:285

...

Stack trace

Here is what happened, the most recent locations are first:

    _similar_shape(itr::Nothing, ::Base.HasLength) @ [array.jl:710](https://github.com/JuliaLang/julia/tree/bd47eca2c8aacd145b6c5c02e47e2b9ec27ab456/base/array.jl#L710)
    _array_for(::Type{Core.MethodInstance}, itr::Nothing, isz::Base.HasLength) @ [array.jl:726](https://github.com/JuliaLang/julia/tree/bd47eca2c8aacd145b6c5c02e47e2b9ec27ab456/base/array.jl#L726)
    _method_instances(f_type::Any, t_type::Any) @ Tricks.jl:136
    #s1#1 @ Tricks.jl:20
    var"#s1#1"(T::Any, ::Any, f::Any, t::Any) @ 
    (::Core.GeneratedFunctionStub)(::UInt64, ::LineNumberNode, ::Any, ::Vararg{Any}) @ boot.jl:602
    content(x::HypertextLiteral.Result) @ convert.jl:91
    show(io::IOContext{IOBuffer}, m::MIME{Symbol("text/html")}, toc::PlutoUI.TableOfContentsNotebook.TableOfContents) @ TableOfContents.jl:502
```

**Comment by benbaz-2** [comment]
I get the same error. If you try copy pasting the contents of the cells in a new notebook, it works fine. I have no idea what's happening. My guess is that it has something to do with Pluto because it usually happens when there is a @ bind or PlutoUI.TableOfContents().

**Comment by MendeBadra** [comment]
I think I have found the solution to the problem. You need to update the PlutoUI package inside Pluto notebook. More specifically, 
![image](https://github.com/user-attachments/assets/2708f518-f701-4394-9664-57f325d84ac7)
here the version is outdated and after clicking the up arrow button and restarting the notebook,
![image](https://github.com/user-attachments/assets/5a4bcc23-09eb-448c-9997-570fa145b964)
this seems to solve the problem.

**Comment by hydrosquall** [comment]
I confirmed @MendeBadra 's fix, namely using PlutoUI version `v0.7.59` instead of `v0.7.48` resolved the error for me. 

It looks like the old version is used from the Fall23 version served by the [older course homepage](https://computationalthinking.mit.edu/Fall23/images_abstractions/images/)

https://github.com/mitmath/computational-thinking/blob/36651ba1903a871acd11e3afc9b179c3b1aaced2/src/images_abstractions/images.jl#L778

But the new version on the Fall24 branch has the correct latest version ( as part of an update for support with julia 1.10 https://github.com/mitmath/computational-thinking/pull/144 ) 

https://github.com/mitmath/computational-thinking/blob/829eea2c0fc7c61a1ca0db332a3ef716a9cf3756/src/images_abstractions/images.jl#L778

**Comment by fonsp** [comment]
This is fixed now, thanks for the report! The problem was caused by the update to Julia 1.10, which had a change to `@generated` that broke HypertextLiteral.jl (https://github.com/JuliaPluto/HypertextLiteral.jl/pull/36)
